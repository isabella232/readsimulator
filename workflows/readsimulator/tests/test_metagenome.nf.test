nextflow_workflow {

    name "Test workflow: READSIMULATOR"
    script "../main.nf"
    workflow "READSIMULATOR"
    tag "workflows"
    tag "readsimulator"
    tag "readsimulator_test_metagenome"

    test("metagenome = true") {

        when {
            workflow {
                """
                input[0] = Channel.of(
                    [ [ id:'first', seed:40] ],
                    [ [ id:'second', seed:41] ],
                    [ [ id:'third', seed:42] ],
                )
                """
            }
            params {
                metagenome          = true
                metagenome_n_reads  = '100K'
                fasta               = 'https://raw.githubusercontent.com/nf-core/test-datasets/readsimulator/testdata/GCF_024334085.1_ASM2433408v1_genomic.fna.gz'
                outdir              = "results"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.get(0).findAll { it != "multiqc_report" }
                ).match() }
            )
        }
    }
}
