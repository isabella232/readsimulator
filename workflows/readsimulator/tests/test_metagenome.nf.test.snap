{
    "metagenome = true": {
        "content": [
            [
                [
                    {
                        "id": "first",
                        "seed": 40,
                        "outdir": "insilicoseq",
                        "datatype": "metagenomic_illumina"
                    },
                    [
                        "first_R1.fastq.gz:md5,2a324062e0afb4d3eab8134479f317d4",
                        "first_R2.fastq.gz:md5,9e0c86c0ff15091734fe7d52039f7727"
                    ]
                ],
                [
                    {
                        "id": "second",
                        "seed": 41,
                        "outdir": "insilicoseq",
                        "datatype": "metagenomic_illumina"
                    },
                    [
                        "second_R1.fastq.gz:md5,3536190205e1a6ba3609fe6a1567e0c7",
                        "second_R2.fastq.gz:md5,a91ff7cf5ea2069d541fc7d081c82bfb"
                    ]
                ],
                [
                    {
                        "id": "third",
                        "seed": 42,
                        "outdir": "insilicoseq",
                        "datatype": "metagenomic_illumina"
                    },
                    [
                        "third_R1.fastq.gz:md5,137d1f3ced4783a0a45f6d7aed6da743",
                        "third_R2.fastq.gz:md5,3a78f893f1c12af70d13fe3df59eaa84"
                    ]
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-19T14:51:11.751345459"
    }
}